clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -323,19 +325,114 @@ struct TemplateParameterListBuilder {
     return *this;
   }
 
-  BuiltinTypeDeclBuilder &finalizeTemplateArgs() {
+  /*
+The concept specialization expression (CSE) constructed in
```

Comment:
nit: generally LLVM does per-line comment tokens for blocks (`//` rather than `/* ... */`)